src/main.js

```diff
@@ -74,135 +85,207 @@ const startServer = (configPath, config) => {
-        console.log(
-          `Make sure ${path.basename(mbtilesFile)} is valid MBTiles.`,
-        );
-        process.exit(1);
-      }
-      const bounds = info.bounds;
+  const styleDir = path.resolve(
+    __dirname,
+    '../node_modules/tileserver-gl-styles/',
+  );
 
-      const styleDir = path.resolve(
-        __dirname,
-        '../node_modules/tileserver-gl-styles/',
-      );
+  const config = {
+    options: {
+      paths: {
+        root: styleDir,
+        fonts: 'fonts',
+        styles: 'styles',
+        mbtiles: inputFilePath,
+        pmtiles: inputFilePath,
+      },
+    },
+    styles: {},
+    data: {},
+  };
 
-      const config = {
-        options: {
-          paths: {
-            root: styleDir,
-            fonts: 'fonts',
-            styles: 'styles',
-            mbtiles: path.dirname(mbtilesFile),
-          },
-        },
-        styles: {},
-        data: {},
-      };
-
-      if (
-        info.format === 'pbf' &&
-        info.name.toLowerCase().indexOf('openmaptiles') > -1
-      ) {
+  const extension = inputFile.split('.').pop().toLowerCase();
+  if (extension === 'pmtiles') {
+    let FileOpenInfo = PMtilesOpen(inputFile);
+    const metadata = await GetPMtilesInfo(FileOpenInfo.pmtiles);
+    if (FileOpenInfo.fd !== undefined) {
+      PMtilesClose(FileOpenInfo.fd);
+    }
+    FileOpenInfo = null;
+
+    if (
+      metadata.format === 'pbf' &&
+      metadata.name.toLowerCase().indexOf('openmaptiles') > -1
+    ) {
+      if (isValidHttpUrl(inputFile)) {
+        config['data'][`v3`] = {
+          pmtiles: inputFile,
+        };
+      } else {
         config['data'][`v3`] = {
-          mbtiles: path.basename(mbtilesFile),
+          pmtiles: path.basename(inputFile),
         };
+      }
 
-        const styles = fs.readdirSync(path.resolve(styleDir, 'styles'));
-        for (const styleName of styles) {
-          const styleFileRel = styleName + '/style.json';
-          const styleFile = path.resolve(styleDir, 'styles', styleFileRel);
-          if (fs.existsSync(styleFile)) {
-            config['styles'][styleName] = {
-              style: styleFileRel,
-              tilejson: {
-                bounds: bounds,
-              },
-            };
-          }
+      const styles = fs.readdirSync(path.resolve(styleDir, 'styles'));
+      for (const styleName of styles) {
+        const styleFileRel = styleName + '/style.json';
+        const styleFile = path.resolve(styleDir, 'styles', styleFileRel);
+        if (fs.existsSync(styleFile)) {
+          config['styles'][styleName] = {
+            style: styleFileRel,
+            tilejson: {
+              bounds: metadata.bounds,
+            },
+          };
         }
+      }
+    } else {
+      console.log(
+        `WARN: PMTiles not in "openmaptiles" format. Serving raw data only...`,
+      );
+      if (isValidHttpUrl(inputFile)) {
+        config['data'][(metadata.id || 'pmtiles').replace(/[?/:]/g, '_')] = {
```

Comment:
This was originally 

```
          (info.id || 'mbtiles')
            .replace(/\//g, '_')
            .replace(/:/g, '_')
            .replace(/\?/g, '_')
```

My assumption was this was to remove "://" from mbtiles:// , is someone accidently used that as the file type identifier. , since the rest of the code had assumed "mbtiles" would be there.

Reply:
but looking again with the underscore, maybe I was wrong on that.